src/world/entity/model.hpp

```diff
@@ -4,19 +4,74 @@
 #include "gui/render/gpu_data/array_buffer.hpp"
 #include "gui/render/gpu_data/gpu_data.hpp"
 #include "gui/render/gpu_data/static_mesh.hpp"
-#include "json/json.h"
+#include "manifest.hpp"
 #include "placement.hpp"
 #include "types.hpp"
 #include "util/voxel.hpp"
 
 #include <filesystem>
+#include <optional>
 #include <unordered_set>
 #include <vector>
 
 namespace world {
 
 namespace entity {
 
+struct global_illumination_t {
+    float ambient;
+    float specular;
+    float diffuse;
+};
+
+// TODO move to cpp
+struct remapping_t {
+    std::unordered_map<ColorInt, ColorInt> map;
+
+    void 
+    read_map(std::unordered_map<std::string, std::string> input)
+    {
+        for (const auto& [k, v] : input)
+            map[std::stoull(k, nullptr, 16)] = std::stoull(v,  nullptr, 16);
+    }
+
+    std::unordered_map<std::string, std::string> 
+    write_map() const 
+    {
+        std::unordered_map<std::string, std::string> res;
+
+        for (const auto& [key, value] : map) {
+            std::stringstream stream_key;
+            stream_key << std::hex << key;
+            std::string str_key(stream_key.str());
+
+            std::stringstream stream_value;
+            stream_value << std::hex << value;
+            std::string str_value(stream_value.str());
+
+            res.insert({str_key, str_value});
```

Comment:
change to use fmt like in the godbolt example

iostream is about 20-30x slower